target_postgres/rdbms_base.py

```diff
@@ -0,0 +1,273 @@
+                item_json_schema['type'].append('null')
+            top_level_schema[next_path] = item_json_schema
+
+
+def _create_subtable(table_name, table_json_schema, key_prop_schemas, subtables, level):
+    if json_schema.is_object(table_json_schema['items']):
+        new_properties = table_json_schema['items']['properties']
+    else:
+        new_properties = {'value': table_json_schema['items']}
+
+    key_properties = []
+    for pk, item_json_schema in key_prop_schemas.items():
+        key_properties.append(SINGER_SOURCE_PK_PREFIX + pk)
+        new_properties[SINGER_SOURCE_PK_PREFIX + pk] = item_json_schema
+
+    new_properties[SINGER_SEQUENCE] = {
+        'type': ['null', 'integer']
+    }
+
+    for i in range(0, level + 1):
+        new_properties[SINGER_LEVEL.format(i)] = {
+            'type': ['integer']
+        }
+
+    new_schema = {'type': ['object'],
+                  'properties': new_properties,
+                  'level': level,
+                  'key_properties': key_properties}
+
+    _denest_schema(table_name, new_schema, key_prop_schemas, subtables, level=level)
+
+    subtables[table_name] = new_schema
+
+
+def _denest_schema(table_name, table_json_schema, key_prop_schemas, subtables, current_path=None, level=-1):
+    new_properties = {}
+    for prop, item_json_schema in table_json_schema['properties'].items():
+        if current_path:
+            next_path = current_path + SEPARATOR + prop
+        else:
+            next_path = prop
+
+        if json_schema.is_object(item_json_schema):
+            not_null = 'null' not in item_json_schema['type']
+            _denest_schema_helper(table_name + SEPARATOR + next_path,
+                                  item_json_schema,
+                                  not_null,
+                                  new_properties,
+                                  next_path,
+                                  key_prop_schemas,
+                                  subtables,
+                                  level)
+        elif json_schema.is_iterable(item_json_schema):
+            _create_subtable(table_name + SEPARATOR + next_path,
+                             item_json_schema,
+                             key_prop_schemas,
+                             subtables,
+                             level + 1)
+        else:
+            new_properties[prop] = item_json_schema
+    table_json_schema['properties'] = new_properties
+
+
+def _flatten_schema(stream_buffer, root_table_name, schema):
+    subtables = {}
+    key_prop_schemas = {}
+    for key in stream_buffer.key_properties:
+        key_prop_schemas[key] = schema['properties'][key]
+    _denest_schema(root_table_name, schema, key_prop_schemas, subtables)
+
+    ret = []
+    for name, schema in subtables.items():
+        ret.append(to_table_schema(name, schema['level'], schema['key_properties'], schema['properties']))
+    return ret
+
+
+class RDBMSNotImplementedError(NotImplementedError):
```

Comment:
This seems unnecessary, it's pretty common to just throw straight up NotImplementedError from a base class function.